### PR TITLE
feat: add ability to use smart importer with investments

### PR DESCRIPTION
**Pull request by patbakdev**
Here is my first attempt at addressing #35 with the intention of making the least amount of code changes and avoid breaking changes. I'm not happy with the duplicative `if target_acct` code, but its the least risky.

- Make 'transfer' account optional
- Don't add posting if target account is missing
- Add optional existing_entries parameter to extract method

I also have a bit more involved code change that rolls the "leaf" formatting into `self.get_target_acct`, which removes the duplication and seems a bit more logical to me, but it would *potentially* break the function signature if done right. You can see the diff here: https://github.com/redstreet/beancount_reds_importers/compare/main...patbakdev:beancount_reds_importers:smart_importer2?expand=1

### CODE REVIEW

File: beancount_reds_importers/libtransactionbuilder/investments.py [hunk 1]
```diff
@@ -136,6 +142,8 @@ def get_target_acct(self, transaction, ticker):
             return target
         if transaction.type == 'income' and getattr(transaction, 'income_type', None) == 'DIV':
             return self.target_account_map.get('dividends', None)
+        if transaction.type in ["other", "credit", "debit", "cash", "dep", "transfer"] and 'transfer' not in self.config:
```
Comment (redstreet) [inline beancount_reds_importers/libtransactionbuilder/investments.py:145]: Is this line necessary? Won't the line below will take care of it?

**Review by redstreet — CHANGES_REQUESTED** [review]
Looks good otherwise.

**Comment by redstreet** [comment]
> I also have a bit more involved code change that rolls the "leaf" formatting into `self.get_target_acct`, which removes the duplication and seems a bit more logical to me, but it would _potentially_ break the function signature if done right. You can see the diff here: https://github.com/redstreet/beancount_reds_importers/compare/main...patbakdev:beancount_reds_importers:smart_importer2?expand=1

The concern with this is `get_target_acct_custom` would be forced to do the leaf formatting. `get_target_acct_custom` is meant to be overridden in custom importers, so I'd like to keep this as simple as possible. Do let me know if that makes sense or if I'm missing something.

File: beancount_reds_importers/libtransactionbuilder/investments.py (hunk 1 above)
Comment (patbakdev) [reply beancount_reds_importers/libtransactionbuilder/investments.py:145]: Yep. I'll remove.

**Comment by patbakdev** [comment]
> The concern with this is get_target_acct_custom would be forced to do the leaf formatting.

I think I handled this, but since the less intrusive fix is to go with what I have here, I am fine with skipping that.

**Comment by redstreet** [comment]
Awesome, thanks for this change (and for the well formatted commit message too)! It's a good feature and a clean implementation 👍 .

**Pull request merged**